include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,226 @@
+                T(-1.0), v, w2, T(1.0), std::move(c));
+}
+
+// Implementation based on:
+// 1. Part of algorithm 6 "LAPACK Algorithm for the eigenvector back-transformation", page 15, PhD thesis
+// "GPU Accelerated Implementations of a Generalized Eigenvalue Solver for Hermitian Matrices with
+// Systematic Energy and Time to Solution Analysis" presented by Raffaele Solcà (2016)
+// 2. Report "Gep + back-transformation", Alberto Invernizzi (2020)
+// 3. Report "Reduction to band + back-transformation", Raffaele Solcà (2020)
+// 4. G. H. Golub and C. F. Van Loan, Matrix Computations, chapter 5, The Johns Hopkins University Press
+template <class T>
+struct BackTransformation<Backend::MC, Device::CPU, T> {
+  static void call_FC(Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+  static void call_FC(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c,
+                      Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mv = mat_v.nrTiles().rows();
+  const SizeType nv = mat_v.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+  const SizeType nb = mat_v.blockSize().cols();
+  const SizeType ms = mat_v.size().rows();
+  const SizeType ns = mat_v.size().cols();
+
+  // Matrix T
+  int tottaus;
+  if (ms < mb || ms == 0 || nv == 0)
+    tottaus = 0;
+  else
+    tottaus = (ms / mb - 1) * mb + ms % mb;
+
+  if (tottaus == 0)
+    return;
+
+  LocalElementSize sizeT(tottaus, tottaus);
+  TileElementSize blockSizeT(mb, mb);
+  Matrix<T, Device::CPU> mat_t(sizeT, blockSizeT);
+
+  Matrix<T, Device::CPU> mat_vv({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w2({mb, mat_c.size().cols()}, mat_c.blockSize());
+
+  SizeType last_mb;
+  if (mat_v.blockSize().cols() == 1) {
+    last_mb = 1;
+  }
+  else if (mat_v.size().cols()) {
+    if (mat_v.size().cols() % mat_v.blockSize().cols() == 0)
+      last_mb = mat_v.blockSize().cols();
+    else
+      last_mb = mat_v.size().cols() % mat_v.blockSize().cols();
+  }
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType last_reflector_idx = mat_v.nrTiles().cols() - 2;

```

Comment:
AFAICT this is not the last reflector index. instead, since you are playing with the number of tiles, it is the index of the last panel (i.e. tile-wide) of reflectors.